specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2024-05-01-preview/appplatform.json

```diff
@@ -15184,6 +15691,285 @@
+          "type": "integer"
+        }
+      }
+    },
+    "JobResourceCollection": {
+      "description": "List of Azure Spring Apps Jobs and a possible link for next set.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of Job resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobResource"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string"
+        }
+      }
+    },
+    "JobExecutionProperties": {
+      "description": "Job's execution template, containing configuration for an execution",
+      "type": "object",
+      "properties": {
+        "template": {
+          "$ref": "#/definitions/JobExecutionTemplate",
+          "description": "The template which is applied for the execution of the Job."
+        },
+        "resourceRequests": {
+          "$ref": "#/definitions/ResourceRequests",
+          "description": "The requested resource quantity for required CPU and Memory."
+        }
+      }
+    },
+    "JobExecutionTemplate": {
+      "description": "Job's execution template, containing configuration for an execution",
+      "type": "object",
+      "properties": {
+        "environmentVariables": {
+          "description": "Environment variables of Job execution",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnvVar"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "args": {
+          "description": "Arguments for the Job execution.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "JobExecution": {
+      "description": "Azure Spring Apps Job execution.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "name": {
+          "description": "Job execution Name.",
+          "type": "string"
+        },
+        "status": {
+          "description": "Current state of the job execution",
+          "enum": [
+            "Running",
+            "Pending",
+            "Canceled",
+            "Failed",
+            "Completed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "JobExecutionRunningState",
+            "modelAsString": true
+          }
+        },
+        "startTime": {
+          "description": "Job execution start time.",
+          "format": "date-time",
+          "type": "string"
+        },
+        "endTime": {
+          "description": "Job execution end time.",
+          "format": "date-time",
+          "type": "string"
+        },
+        "properties": {
+          "description": "Job's execution properties.",
+          "$ref": "#/definitions/JobExecutionProperties"
```

Comment:
What is the difference between the `job.properties.template.env`, `jobExecution.properties.template.env` and `jobExecution.jobSnapshot.template.env`? as well as the args...

Reply:
One follow question, what if the execution only return the snapshot with applied data. Then from API version, it is clear to show what is the env, arg the execution really take, we don't need to explain the priority.

Reply:
As offline synced, will return the real applied data from backend. No change in swagger model.

Reply:
I am confused about that, why the swagger model no need to be changed?